cmd/store/import.go

```diff
@@ -35,79 +36,156 @@ import (
+	err = authModel.ReadModelFromString(storeData.Model, format)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read model: %w", err)
+	}
 
-		err = authModel.ReadModelFromString(storeData.Model, format)
+	modelWriteRes, err := model.Write(fgaClient, authModel)
+	if err != nil {
+		return nil, fmt.Errorf("failed to write model: %w", err)
+	}
+
+	response.Model = modelWriteRes
+
+	return response, nil
+}
+
+func importStore(
+	clientConfig fga.ClientConfig,
+	fgaClient client.SdkClient,
+	storeData *storetest.StoreData,
+	format authorizationmodel.ModelFormat,
+	storeID string,
+	maxTuplesPerWrite int,
+	maxParallelRequests int,
+	fileName string,
+) (*CreateStoreAndModelResponse, error) {
+	var (
+		response *CreateStoreAndModelResponse
+		err      error
+	)
+
+	if storeID == "" {
+		response, err = createStore(clientConfig, storeData, format, fileName)
 		if err != nil {
-			return nil, err //nolint:wrapcheck
+			return nil, fmt.Errorf("failed to create store: %w", err)
 		}
-
-		modelWriteRes, err := model.Write(fgaClient, authModel)
+	} else {
+		response, err = updateStore(clientConfig, fgaClient, storeData, format, storeID)
 		if err != nil {
-			return nil, fmt.Errorf("failed to write model due to %w", err)
+			return nil, fmt.Errorf("failed to update store: %w", err)
 		}
-
-		response.Model = modelWriteRes
 	}
 
-	fgaClient, err := clientConfig.GetFgaClient()
+	fgaClient, err = clientConfig.GetFgaClient()
 	if err != nil {
-		return nil, fmt.Errorf("failed to initialize FGA Client due to %w", err)
+		return nil, fmt.Errorf("failed to initialize FGA Client: %w", err)
 	}
 
 	writeRequest := client.ClientWriteRequest{
 		Writes: storeData.Tuples,
 	}
 
-	_, err = tuple.ImportTuples(fgaClient, writeRequest, maxTuplesPerWrite, maxParallelRequests)
-	if err != nil {
-		return nil, err //nolint:wrapcheck
+	// Initialize progress bar
+	bar := progressbar.NewOptions(len(storeData.Tuples),
+		progressbar.OptionSetDescription("Importing tuples"),
+		progressbar.OptionShowCount(),
+		progressbar.OptionSetWidth(progressBarWidth),
+		progressbar.OptionClearOnFinish(),
+		progressbar.OptionFullWidth(),
+		progressbar.OptionThrottle(65*progressBarSleepDelay), // Adjust this as needed
+		progressbar.OptionShowIts(),
+		progressbar.OptionSetItsString("tuples"),
+		progressbar.OptionSetPredictTime(true),
+		progressbar.OptionSetRenderBlankState(true),
+		progressbar.OptionSetTheme(progressbar.Theme{
+			Saucer:        "#",
+			SaucerPadding: " ",
+			BarStart:      "[",
+			BarEnd:        "]",
+		}),
+	)
+
+	for index := 0; index < len(storeData.Tuples); index += maxTuplesPerWrite {
+		end := index + maxTuplesPerWrite
+		if end > len(storeData.Tuples) {
+			end = len(storeData.Tuples)
+		}
+
+		writeRequest.Writes = storeData.Tuples[index:end]
+		_, err = tuple.ImportTuples(fgaClient, writeRequest, maxTuplesPerWrite, maxParallelRequests)
+		if err != nil {
+			return nil, fmt.Errorf("failed to import tuples: %w", err)
+		}
```

Comment:
We can skip this lint check (or we can it in ignore) since it's a known problem - https://github.com/golangci/golangci-lint/issues/1510


<img width="873" alt="image" src="https://github.com/openfga/cli/assets/55068936/b5eb08a0-d2c3-4dfa-932b-7aa02e4b5428">

Reply:
Fixed in [`7006ccf`](https://github.com/openfga/cli/pull/348/commits/7006ccfd342edad2b7c716127358cc43dc91d72f)